@xen-orchestra/lite/src/components/Table.tsx

```diff
@@ -19,55 +54,320 @@ interface ParentState {}
+      },
+      toggleAllSelectedItem: function () {
+        if (this.props.collection === undefined) {
+          return
+        }
+        this.props.onSelectItems?.(
+          this.props.selectedItems.length !== this.props.collection.length ? this.props.collection : []
+        )
+      },
+      toggleSelectItem: function (item: any) {
+        const { selectedItems } = this.props
+        if (selectedItems === undefined) {
+          return
+        }
+        const index = findIndex(selectedItems, item)
+        index === -1 ? selectedItems.push(item) : selectedItems.splice(index, 1)
+        this.props.onSelectItems(selectedItems)
+      },
+    },
+    computed: {
+      page: (_, { location, stateUrlParam }) =>
+        Number(location.search.match(new RegExp(stateUrlParam + '_page=\\d+'))?.[0]?.split('=')[1] ?? 0),
+      paginatedCollection: ({ page, rowsPerPage }, { collection }) =>
+        collection?.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage),
+      rowsPerPage: (_, { location, stateUrlParam, rowsPerPageOptions }) => {
+        const _rowsPerPageOptions = rowsPerPageOptions ?? [10, 25, 50, 100]
+        let rowsPerPage = Number(
+          location.search.match(new RegExp(stateUrlParam + '_row=\\d+'))?.[0]?.split('=')[1] ?? _rowsPerPageOptions[0]
+        )
+        if (!_rowsPerPageOptions.includes(rowsPerPage)) {
+          console.error('Not allowed rowsPerPage', rowsPerPage)
+          rowsPerPage = _rowsPerPageOptions[0]
+        }
+        return rowsPerPage
+      },
+    },
+  },
+  ({
+    collection,
+    columns,
+    dataType,
+    effects,
+    isItemSelectable,
+    rowsPerPageOptions,
+    selectedItems,
+    state: { page, paginatedCollection, rowsPerPage },
+  }) =>
+    paginatedCollection !== undefined && collection !== undefined ? (
+      <>
+        <Pagination
+          dataType={dataType}
+          nbItemsOnThePage={paginatedCollection.length}
+          nbSelectedItems={selectedItems?.length}
+          nbTotalItems={collection.length}
+          nbTotalPage={Math.ceil(collection.length / rowsPerPage)}
+          onPaginationChange={effects.handlePaginationChange}
+          onShowByChange={effects.handleRowPerPage}
+          page={page}
+          reverse
+          rowPerPageOptions={rowsPerPageOptions}
+          showByValue={rowsPerPage}
+        />
+        <MUITable>
+          <TableHead>
+            <TableRow>
+              {isItemSelectable && (
+                <StyledTableCell padding='checkbox'>
+                  <Checkbox
+                    checked={selectedItems.length > 0 && selectedItems.length === collection.length}
+                    color='primary'
+                    indeterminate={selectedItems.length > 0 && selectedItems.length < collection.length}
+                    onChange={effects.toggleAllSelectedItem}
+                  />
+                </StyledTableCell>
+              )}
```

Comment:
What about selecting all the items of the table, even those that are on other pages?